### PR TITLE
Feature/deep link handler

### PR DESCRIPTION
**Pull request by poocart**
https://www.pivotaltracker.com/story/show/163645691

This PR covers:
1. Deep link handling within the app.
2. Deep link listener for Discourse login integration.
3. Routing fix for "Home" tab: duplicate constant for "Home" route and "Home" tab name.

### CODE REVIEW

File: src/actions/deepLinkActions.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  return async (dispatch: Function) => {
+    dispatch({ type: RESET_DEEP_LINK_DATA });
+  };
+};
+
+export const approveLoginAttemptAction = (loginAttemptToken: string) => {
+  return async (dispatch: Function, getState: Function, api: Object) => {
+    try {
+      const result = await api.approveLoginToExternalResource(loginAttemptToken);
+      if (result !== undefined
```
Comment (jsidorenko) [inline src/actions/deepLinkActions.js]: it's nearly the same as `if (result && !result.error) {`

File: src/actions/deepLinkActions.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        && result.error === undefined) {
+        dispatch({ type: RESET_DEEP_LINK_DATA });
+      } else {
+        throw new Error();
```
Comment (jsidorenko) [inline src/actions/deepLinkActions.js]: It's better to remove `if (){ }else{ }` and throw an error after the api call:
`if (!result || result.error) throw new Error();`

Comment (poocart) [reply src/actions/deepLinkActions.js]: Good _catch_ 👍

File: src/screens/Home/Home.js [hunk 3]
```diff
@@ -432,6 +447,9 @@ class HomeScreen extends React.Component<Props, State> {
     const hasIntercomNotifications = !!intercomNotificationsCount;
     const isWalletBackedUp = isImported || isBackedUp;
 
+    const { loginAttemptToken } = deepLinkData;
+    const showLoginApproveModal = loginAttemptToken !== null && loginAttemptToken !== undefined;
```
Comment (jsidorenko) [inline src/screens/Home/Home.js]:
```suggestion
    const showLoginApproveModal = !!loginAttemptToken;
```

**Pull request merged**